core/src/main/java/org/axonframework/config/AggregateConfigurer.java

```diff
@@ -127,11 +132,13 @@ protected AggregateConfigurer(Class<A> aggregate) {
         return configurer.configureRepository(
                 c -> new GenericJpaRepository<>(c.getComponent(EntityManagerProvider.class,
                                                                () -> {
-                                                                   throw new AxonConfigurationException(format("JPA has not been correctly configured for aggregate [%s]. Either provide an EntityManagerProvider, or use DefaultConfigurer.jpaConfiguration(...) to define one for the entire configuration.",
-                                                                                                               aggregateType.getSimpleName()));
-
+                                                                   throw new AxonConfigurationException(format(
+                                                                           "JPA has not been correctly configured for aggregate [%s]. Either provide an EntityManagerProvider, or use DefaultConfigurer.jpaConfiguration(...) to define one for the entire configuration.",
+                                                                           aggregateType.getSimpleName()));
                                                                }),
-                                                configurer.metaModel.get(), c.eventBus()));
+                                                configurer.metaModel.get(),
+                                                c.eventBus(),
+                                                (RepositoryProvider) c::repository));
```

Comment:
Why do you need to cast the lambda here, whilst you do not have to do that for the previous two `c::repository` calls?